### PR TITLE
[Serving] Support default env var override behavior for connection

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_utils.py [hunk 1]
```diff
@@ -317,6 +317,28 @@ def _match_env_reference(val: str):
...
+    for connection_name, connection in connections.items():
+        values = connection.get("value", {})
+        for key, val in values.items():
+            if key in connection.get("secret_keys", []):
+                continue
+            connection_name = connection_name.replace(" ", "_")
+            env_name = f"{connection_name}_{key}".upper()
+            if env_name not in os.environ:
+                continue
+            values[key] = os.environ[env_name]
```
Comment (brynn-code) [inline src/promptflow/promptflow/_sdk/_utils.py:338]: Add log info if we replaced keys/secrets like "Connection {name} {key} overrides with environment variable {env var name}"

File: src/promptflow/promptflow/_sdk/_utils.py [hunk 2]
```diff
@@ -317,6 +317,28 @@ def _match_env_reference(val: str):
...
+    is a custom connection named 'custom_connection' with a configuration key called 'chat_deployment_name,' the
+    function will attempt to retrieve 'chat_deployment_name' from the environment variable
+    'CUSTOM_CONNECTION_CHAT_DEPLOYMENT_NAME' by default. If the environment variable is not set, it will use the
+    original value as a fallback.
+    """
+
+    for connection_name, connection in connections.items():
+        values = connection.get("value", {})
+        for key, val in values.items():
+            if key in connection.get("secret_keys", []):
```
Comment (brynn-code) [inline src/promptflow/promptflow/_sdk/_utils.py:332]: Do not need this condition, as we support override both configs & secrets.

File: src/promptflow/promptflow/_sdk/_utils.py (hunk 1 above)
Comment (YingChen1996) [reply src/promptflow/promptflow/_sdk/_utils.py:336]: added

File: src/promptflow/promptflow/_sdk/_utils.py (hunk 2 above)
Comment (YingChen1996) [reply src/promptflow/promptflow/_sdk/_utils.py]: updated, I thought connection configs only contain "configs" field.

File: docs/how-to-guides/deploy-a-flow/deploy-using-dev-server.md [hunk 3]
```diff
@@ -14,6 +14,11 @@ an example to show how to deploy a flow.
 Please ensure you have [create the connection](../manage-connections.md#create-a-connection) required by flow, if not, you could
 refer to [Setup connection for web-classification](https://github.com/microsoft/promptflow/tree/main/examples/flows/standard/web-classification).
 
+Note: We will use relevant environment variable to override connection configurations in serving mode. For instance,
```
Comment (brynn-code) [inline docs/how-to-guides/deploy-a-flow/deploy-using-dev-server.md:17]: We will use relevant environment variable ({connection_name}_{key_name}) to override connection configurations in serving mode, white space in connection name will be removed directly from environment variable name.

**Review by brynn-code — DISMISSED** [review]

File: docs/how-to-guides/deploy-a-flow/deploy-using-dev-server.md (hunk 3 above)
Comment (YingChen1996) [reply docs/how-to-guides/deploy-a-flow/deploy-using-dev-server.md]: updated

**Review by brynn-code — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Pull request merged**